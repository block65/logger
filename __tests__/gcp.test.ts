import { beforeEach, describe, expect, jest, test } from '@jest/globals';
import { createLoggerWithWaitableMock } from './helpers.js';

describe('GCP', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers('modern');
    jest.setSystemTime(new Date('2009-02-13T23:31:30.000Z'));
  });

  test('Cloud Run', async () => {
    const [logger, callback] = await createLoggerWithWaitableMock({
      platform: 'gcp-cloudrun',
    });

    logger.warn(new Error('hello'));
    await expect(callback.waitUntilCalled()).resolves.toMatchSnapshot();
  });

  test('Cloud Run Error Object', async () => {
    const [logger, callback] = await createLoggerWithWaitableMock({
      platform: 'gcp-cloudrun',
    });

    logger.error(new Error('Ded 1'));

    await expect(callback.waitUntilCalled()).resolves.toMatchSnapshot();
  });

  test('Cloud Run Fatal with Error Object', async () => {
    const [logger, callback] = await createLoggerWithWaitableMock({
      platform: 'gcp-cloudrun',
    });

    logger.fatal(new Error(`Ded 2`));

    await expect(callback.waitUntilCalled()).resolves.toMatchSnapshot();
  });
});
