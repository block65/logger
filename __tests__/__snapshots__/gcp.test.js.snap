// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GCP Cloud Run 1`] = `
Array [
  Object {
    "message": "hello",
    "severity": "WARNING",
    "stack_trace": "Error: hello
    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/gcp.test.js:13:21)",
    "time": "2009-02-13T23:31:30.000Z",
  },
]
`;

exports[`GCP Cloud Run Error Object 1`] = `
Array [
  Object {
    "@type": "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent",
    "message": "Ded 1",
    "severity": "ERROR",
    "stack_trace": "Error: Ded 1
    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/gcp.test.js:20:22)",
    "time": "2009-02-13T23:31:30.000Z",
  },
]
`;

exports[`GCP Cloud Run Fatal with Error Object 1`] = `
Array [
  Object {
    "@type": "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent",
    "message": "Ded 2",
    "severity": "CRITICAL",
    "stack_trace": "Error: Ded 2
    at Object.<anonymous> (~/block65_logger/bazel-out/k8-fastbuild/~/block65_logger/__tests__/gcp.test.js:27:22)",
    "time": "2009-02-13T23:31:30.000Z",
  },
]
`;
